### PR TITLE
[NG-2100] Update Dependencies

### PR DESCRIPTION
**Pull request by limtingzhi**
**Changes:**
- Upgrade dependencies
- Major dependencies changes
  - `conventional-changelog/commitlint` related packages (should be nothing much for us)
    - https://github.com/conventional-changelog/commitlint/releases/tag/v14.0.0
    - https://github.com/conventional-changelog/commitlint/releases/tag/v15.0.0
    - https://github.com/conventional-changelog/commitlint/releases/tag/v16.0.0
  - `semantic-release/semantic-release` related packages
    - https://github.com/semantic-release/semantic-release/releases/tag/v18.0.0 (minimum required node version is v14.17)
  - `motdotla/dotenv` (breaking changes: drop support for Node v10)
    - https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md#1100-2022-01-11
- Ignore major dependencies updates for dependabot config
- Add pull request template

**JIRA Tickets:**
- Update Dependabot Config To Update Only Minor Deps (https://nugitco.atlassian.net/browse/NG-2100)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,13 +24,16 @@
     "url": "https://github.com/nugit/actions-pr-changelog/issues"
   },
   "homepage": "https://github.com/nugit/actions-pr-changelog#readme",
+  "engines": {
+    "node": ">=14.17"
+  },
```
Comment (moroine) [inline package.json:29]: Not sure if this is a breaking change. Because the action must run in Node14+ ?

Comment (limtingzhi) [reply package.json:29]: @moroine Hmmm, I'm not sure too. I can update it to `12`, but this repo needs `14` when developing. I think it is alright.

What do you think?

Comment (moroine) [reply package.json:29]: I think you can mark a new major version no?

Comment (limtingzhi) [reply package.json:29]: Can also 👍🏻

**Review by moroine — APPROVED** [review]

**Pull request merged**